crates/resctrl/src/lib.rs

```diff
@@ -905,4 +964,96 @@ mod tests {
             other => panic!("expected capacity, got {:?}", other),
         }
     }
+
+    #[test]
+    fn test_reconcile_group_converges() {
+        let mut fs = MockFs::default();
+        // Simulate mounted under default root
+        fs.add_file(
+            Path::new("/proc/mounts"),
+            "resctrl /sys/fs/resctrl resctrl rw 0 0\n",
+        );
+        let root = PathBuf::from("/sys/fs/resctrl");
+        fs.add_dir(&root);
+
+        // Create group and its tasks file
+        let group_path = root.join("pod_abc");
+        fs.add_dir(&group_path);
+        let tasks = group_path.join("tasks");
+        fs.add_file(&tasks, "");
+
+        let rc = Resctrl::with_provider(
+            fs,
+            Config {
+                root: root.clone(),
+                group_prefix: "pod_".into(),
+                auto_mount: false,
+            },
+        );
+
+        // Desired PIDs remain stable; should converge in <= 2 passes
```

Comment:
can we add more variants:
1. in the first few iterations the set changes (requiring more iterations) but eventually it stops changing so we get successful convergence
2. the set continuously changes so there is no convergence
3. PIDs are added to the pid source, but those same PIDs are added to the resctrl tasks (simulating that PIDs already in the resctrl group fork -- the children will be in the cgroup and resctrl both)

let's try to minimize repetitive code